R/formula.R

```diff
@@ -0,0 +1,129 @@
+new_googlesheets4_formula <- function(x = character()) {
+  vec_assert(x, character())
+  new_vctr(x, class = "googlesheets4_formula")
+}
+
+#' `googlesheets4_formula` class
+#'
+#' In order to write a formula into Google Sheets, you need to store it as an
+#' object of class `googlesheets4_formula`. This is how we distinguish a
+#' "regular" character string from a string that should be interpreted as a
+#' formula. `googlesheets4_formula` is an S3 class implemented using the [vctrs
+#' package](https://vctrs.r-lib.org/articles/s3-vector.html).
+#'
+#' @param x Character.
+#'
+#' @return An S3 vector of class `googlesheets4_formula`.
+#' @export
+#' @family write functions
+#'
+#' @examples
+#' if (sheets_has_token()) {
+#' dat <- data.frame(X = c(1, 5, 3, 2, 4, 6))
+#'
+#' ss <- sheets_write(dat)
+#'
+#' summaries <- tibble::tribble(
+#'        ~ desc, ~ summaries,
+#'         "max", "=max(A:A)",
+#'         "sum", "=sum(A:A)",
+#'         "min", "=min(A:A)",
+#'   "sparkline", "=SPARKLINE(A:A, {\"color\", \"blue\"})"
+#' )
+#'
+#' # explicitly declare a column as `googlesheets4_formula`
+#' summaries$summaries <- googlesheets4_formula(summaries$summaries)
+#'
+#' sheets_edit(ss, data = summaries, range = "C1", reformat = FALSE)
+#'
+#' miscellany <- tibble::tribble(
+#'        ~ desc, ~ example,
+#'   "hyperlink", "=HYPERLINK(\"http://www.google.com/\",\"Google\")",
+#'        "image", "=IMAGE(\"https://www.google.com/images/srpr/logo3w.png\")"
+#' )
+#' miscellany$example <- googlesheets4_formula(miscellany$example)
+#'
+#' sheets_write(miscellany, ss = ss)
+#'
+#' # clean up
+#' googledrive::drive_trash(ss)
+#' }
+googlesheets4_formula <- function(x = character()) {
+  x <- vec_cast(x, character())
+  new_googlesheets4_formula(x)
+}
+
+#' @importFrom methods setOldClass
+methods::setOldClass(c("googlesheets4_formula", "vctrs_vctr"))
+
+#' @export
+vec_ptype_abbr.googlesheets4_formula <- function(x, ...) {
+  "fmla"
+}
+
+#' non-user-facing vctrs methods for googlesheets4_formula
```

Comment:
```suggestion
#' Interval vctrs methods for googlesheets4_formula
```

Reply:
I'm going to assume this should be "Internal" 😀